### PR TITLE
[FIX] lower priority to use SPM function to deal with json files

### LINKED ISSUE #176: jsonwrite for outdated SPM
The interface of `spm_jsonwrite` changed in SPM in revision r. 6975, making it unusable for outdated SPM installations. 
Installation of JSONio do not change anything as the check for existence for `spm_jsonwrite` comes first.

I see two solution for this (other from keep SPM updated):

- use only JSONio, cite it as dependency
- Invert if in `+bids/+util/jsonencode.m` line 33, so JSONio takes priority

Tell me that you think

UPD:
You also check for native MATLAB jsonencode, but it has the lower priority. May be it can be rendered as highest one

### PR DESCRIPTION
**Pull request by Remi-Gau**
fixes #176 

- [x] change order of priority to check which function to use to deal with JSON: built-in functions > JSONio > SPM
- [x] reformat help sections of those functions

Once merged I will also apply this fix on the `dev` branch.

### CODE REVIEW

File: +bids/+util/jsonencode.m [hunk 1]
```diff
@@ -59,8 +76,21 @@
       fprintf(fid, '%s', txt);
```
Comment (nbeliy) [inline +bids/+util/jsonencode.m:76]: Will it write to file with new lines and indend?

Comment (Remi-Gau) [reply +bids/+util/jsonencode.m:76]: I just checked and if you rely on Matlab built-in function there will not be any JSON "formating".

I might be wrong but I don't think those options are supported by Matlab's functions. :sob:

Comment (gllmflndn) [reply +bids/+util/jsonencode.m:76]: My little finger told me to tell you to wait a few weeks for `PrettyPrint`.

Comment (gllmflndn) [reply +bids/+util/jsonencode.m:76]: https://www.mathworks.com/help/matlab/ref/jsonencode.html#bvb7dgc-2

**Pull request merged**